ctapipe_io_magic/__init__.py

```diff
@@ -645,18 +681,21 @@ def prepare_subarray_info(self):
             u.GHz
         )
         camera_readout = CameraReadout(
-            camera_name='MAGICCam',
+            name='MAGICCam',
             sampling_rate=sampling_speed,
             reference_pulse_shape=pulse_shape,
-            reference_pulse_sample_width=u.Quantity(0.5, u.ns)
+            reference_pulse_sample_width=u.Quantity(0.5, u.ns),
+            n_channels=1,
+            n_pixels=N_PIXELS,
+            n_samples=N_SAMPLES,
```

Comment:
n_pixels is tricky because we have some "dummy" pixels not used for analysis, but n_samples I think you can read directly from RunHeaders rather than define as a constant. All the joint data are 50 samples, but old data (that in principle one could also try to process with ctapipe_io_magic) is with 60 or 80 samples. It might also happen that we will change it in the future (to decrease the dead zone of the hardware stereo trigger)

Reply:
Anyway, CameraReadout is used only to tell how the camera is, if some pixels are excluded from analysis I think this needs to be done later in the event source

Reply:
Now reading the number of samples from RunHeaders (see a89ab9a)